### PR TITLE
aws-janitor: Support for CloudFormation Stacks

### PR DESCRIPTION
**Pull request by akutz**
This patch introduces support for the removal of CloudFormation stacks by the AWS janitor.

/assign @detiber @krzyzacy 
/cc @randomvariable @vincepri

### CODE REVIEW

File: maintenance/aws-janitor/resources/cloud_formation_stacks.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+func (p cloudFormationStack) ResourceKey() string {
+	return p.name
+}
+
+func (p cloudFormationStack) delete(svc *cf.CloudFormation) error {
+	input := &cf.DeleteStackInput{StackName: aws.String(p.name)}
+	if _, err := svc.DeleteStack(input); err != nil {
+		return err
```
Comment (vincepri) [inline maintenance/aws-janitor/resources/cloud_formation_stacks.go]: Should we wait for the stack to be deleted?

Comment (akutz) [reply maintenance/aws-janitor/resources/cloud_formation_stacks.go]: Hi @vincepri,

We're not going to get hung up on waiting for things at this point. There are enough race conditions that I'm going to update the Boskos janitor to do three sweeps with a pause in between each one. 

A separate PR at a later date will work out ordering, waiting, and dependency graphs.

**Comment by krzyzacy** [comment]
/approve
/hold

feel free to `/hold cancel`

**Comment by akutz** [comment]
Well I’m good with it since it does work from my local tests. As @vincepri mentioned, we do need to revisit the resources list and work out a better dependency graph, but that’s phase two.

/hold cancel

File: maintenance/aws-janitor/resources/cloud_formation_stacks.go (hunk 1 above)
Comment (akutz) [reply maintenance/aws-janitor/resources/cloud_formation_stacks.go]: Hi @vincepri,

FWIW, [this](https://github.com/kubernetes/test-infra/pull/11511/files#diff-0e9b3813c170d5dc6a3ef2d4463f3235) is the change to the Boskos AWS janitor that performs, by default, three sweeps.

**Pull request merged**